### ISSUE TITLE
False positive with `unicorn/explicit-length-check`

### ISSUE THREAD

**Issue by Cherry** [sindresorhus/eslint-plugin-unicorn#980]
When getting the length of an unknown array, this rule seems to throw a false positive. In the below contrived example, you can see a function that returns an array, and then a variable assigned to either the length of the `foods` returned, or 0. In reality, this would be a 3rd-party API call where the result may be undefined or falsey.

```js
function getFoods(){
	// in reality, this would be async and come from an API or other file where it's not known if the return result is an array, falsey, or the length
	// here, we'll just return an array of foods
	return ['apple', 'orange', 'lemon'];
}

const foods = getFoods();
const numOfFoods = foods && foods.length || 0; // this throws unicorn/explicit-length-check
```

I can work around it by doing something like:
```js
const foods = getFoods();
let numOfFoods = 0;
if(foods && Array.isArray(foods)){
	numOfFoods = foods.length;
}
```
Is this the intention for clearer code?

**Comment by fisker** [comment]
When I add this, I was thinking `foo || bar.length` should mean to use as a `boolean`.

And for `const x = foo.length || bar`, yes I mean to write in your way(or a  ternary).

However, it seems I forgot `const x = foo.length ?? bar`, this one should pass.

**Comment by fregante** [comment]
Duplicate of https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1840
